src/main/java/uk/gov/justice/hmpps/prison/service/SchedulesService.java

```diff
@@ -181,8 +185,16 @@ public List<PrisonerSchedule> getActivitiesAtAllLocations(final String agencyId,
     }
 
     @VerifyAgencyAccess(overrideRoles = {"SYSTEM_USER", "GLOBAL_SEARCH"})
-    public PrisonerActivitiesCount getCountActivities(String agencyId, LocalDate fromDate, LocalDate toDate, List<TimeSlot> timeSlots) {
-        return prisonerActivitiesCountRepository.getCountActivities(agencyId, fromDate, toDate, timeSlots.stream().map(Enum::name).toList());
+    public PrisonerActivitiesCount getCountActivities(String agencyId, LocalDate fromDate, LocalDate toDate, Set<TimeSlot> timeSlots, Map<Long, Long> attendanceCounts) {
+        final var activities = prisonerActivitiesCountRepository.getActivities(agencyId, fromDate, toDate);
+        final var activitiesInTimeSlots = activities.stream()
+            .filter(p -> CalcDateRanges.eventStartsInTimeslots(p.getStartTime(), timeSlots)).toList();
+
+        final var total = activitiesInTimeSlots.size();
+        final var suspended = activitiesInTimeSlots.stream().filter(p -> "Y".equals(p.getSuspended())).count();
+        final var bookingIds = activitiesInTimeSlots.stream().map(PrisonerActivity::getBookingId).collect(Collectors.toSet());
+        final var relevantAttendanceCounts = attendanceCounts.entrySet().stream().filter(a -> bookingIds.contains(a.getKey())).map(Entry::getValue).reduce(0L, Long::sum);
```

Comment:
This replicates the existing functionality that exists in whereabouts.
We only want to count the attendances if the booking is in the scheduled list.  This means that prisoners who attended at the time but don't have an active booking any more are excluded from the not recorded stats.